include/envoy/ssl/handshaker.h

```diff
@@ -37,9 +37,13 @@ class Handshaker {
   /**
    * Do the handshake.
    *
+   * NB:
    *  * |state| is a mutable reference.
-   *  * |callbacks| may not exist throughout the lifetime of the Handshaker, and
-   *    should not be stored in an implementation.
+   *  * |callbacks| is only expected to exist for the duration of the
```

Comment:
So given this lifetime of `callbacks`, I don't think there's any way to complete a handshake via an async operation. Or am I missing something?

Reply:
Done.

Reply:
Discussed offline. Moved HandshakerCallbacks so that it's now a one-time argument of setCallbacks(), which resolves this lifetime issue.